demos/custom-dialogs/custom-dialog.js

```diff
@@ -74,90 +81,91 @@ CustomDialog.hide = function() {
+        'background-color: #fff;' +
+        'width: 400px;' +
+        'margin: 20px auto 0;' +
+        'padding: 10px;';
+    backdropDiv.appendChild(dialogDiv);
+
+    dialogDiv.onclick = function(event) {
+      event.stopPropagation();
     };
 
-    var dialogInput = document.getElementById('dialogInput');
-    CustomDialog.inputField = dialogInput;
-    if (dialogInput) {
-        dialogInput.focus();
-
-        dialogInput.onkeyup = function(event) {
-            if (event.keyCode == 13) {
-                // Process as OK when user hits enter.
-                onOkay();
-                return false;
-            } else if (event.keyCode == 27)  {
-                // Process as cancel when user hits esc.
-                onCancel();
-                return false;
-            }
-        };
-    } else {
-        var okay = document.getElementById('dialogOkay');
-        okay && okay.focus();
-    }
-
-    if (options.showOkay) {
-        document.getElementById('dialogOkay').onclick = onOkay;
-    }
-    if (options.showCancel) {
-        document.getElementById('dialogCancel').onclick = onCancel;
-    }
-
-    backdropDiv.onclick = onCancel;
-}
+    CustomDialog.backdropDiv_ = backdropDiv;
+    CustomDialog.dialogDiv_ = dialogDiv;
+  }
+  backdropDiv.style.display = 'block';
+  dialogDiv.style.display = 'block';
+
+  dialogDiv.innerHTML =
+      '<header class="customDialogTitle"></header>' +
+      '<p class="customDialogMessage"></p>' +
+      (options.showInput? '<div><input id="customDialogInput"></div>' : '') +
+      '<div class="customDialogButtons">' +
+      (options.showCancel ? '<button id="customDialogCancel">Cancel</button>': '') +
+      (options.showOkay ? '<button id="customDialogOkay">OK</button>': '') +
+      '</div>';
+  dialogDiv.getElementsByClassName('customDialogTitle')[0]
+      .appendChild(document.createTextNode(title));
+  dialogDiv.getElementsByClassName('customDialogMessage')[0]
+      .appendChild(document.createTextNode(message));
+
+  var onOkay = function(event) {
+    CustomDialog.hide();
+    options.onOkay && options.onOkay();
+    event && event.stopPropagation();
+  };
+  var onCancel = function(event) {
+    CustomDialog.hide();
+    options.onCancel && options.onCancel();
+    event && event.stopPropagation();
+  };
+
+  var dialogInput = document.getElementById('customDialogInput');
+  CustomDialog.inputField = dialogInput;
+  if (dialogInput) {
+    dialogInput.focus();
+
+    dialogInput.onkeyup = function(event) {
+      if (event.keyCode == 13) {
+        // Process as OK when user hits enter.
+        onOkay();
+        return false;
+      } else if (event.keyCode == 27)  {
+        // Process as cancel when user hits esc.
+        onCancel();
+        return false;
+      }
+    };
+  } else {
+    var okay = document.getElementById('customDialogOkay');
+    okay && okay.focus();
+  }
+
+  if (options.showOkay) {
+    document.getElementById('customDialogOkay').onclick = onOkay;
```

Comment:
It doesn't matter in this case, but onclick is not the best way to set an event as it only allows a single event per element.  The recommended method is:
document.getElementById('customDialogOkay').addEventListener('click', onOkay);